lms/models/grouping.py

```diff
@@ -23,6 +23,12 @@ class Type(str, Enum):
     __table_args__ = (
         sa.UniqueConstraint("application_instance_id", "authority_provided_id"),
         sa.UniqueConstraint("lms_id", "application_instance_id", "parent_id", "type"),
+        sa.UniqueConstraint("id", "application_instance_id"),
```

Comment:
It was necessary to add this `(id, application_instance_id)` unique constraint to get around this error:

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InvalidForeignKey) there is no unique constraint matching given keys for referenced table "grouping"
```

It seems a bit odd because `id` is already unique (it's the primary key) so `(id, application_instance_id)` must be unique anyway, but apparently a separate unique constraint is needed.